commands/deploy_test.go

```diff
@@ -42,6 +52,125 @@ func Test_deploy(t *testing.T) {
 	}
 }
 
+func Test_overrideFlag_deployToNamespace(t *testing.T) {
+	var request types.FunctionDeployment
+	envMap := make(map[string]string)
+	function := make(map[string]stack.Function)
+
+	tests := []struct {
+		stackNamespace    string
+		flagNamespace     string
+		expectedNamespace string
+	}{
+		// Test cases
+		{
+			stackNamespace:    "openfaas-fn",
+			flagNamespace:     "default",
+			expectedNamespace: "default",
+		},
+
+		{
+			stackNamespace:    "",
+			flagNamespace:     "default",
+			expectedNamespace: "default",
+		},
+
+		{
+			stackNamespace:    "default",
+			flagNamespace:     "",
+			expectedNamespace: "default",
+		},
+
+		{
+			stackNamespace:    "",
+			flagNamespace:     "",
+			expectedNamespace: "openfaas-fn",
+		},
+	}
+
+	// Pull templates for faas-cli deploy to work
+	faasCmd.SetArgs([]string{
+		"template",
+		"pull",
+	})
+	defer os.RemoveAll("./template")
+	defer os.RemoveAll("./testdata")
+	err := faasCmd.Execute()
+	if err != nil {
+		t.Fatalf("Error pulling in the templates:\n%v", err)
+	}
+
+	// Run the test for each test case defined in "tests"
+	for _, testCase := range tests {
+		testCase := testCase
+
+		// Populating the schema for the stack yaml
+		function["nodeinfo"] = stack.Function{
+			Name:        "nodeinfo-http",
+			Language:    "node12",
+			Handler:     "./nodeinfo-http",
+			Image:       "functions/nodeinfo-http:latest",
+			Namespace:   testCase.stackNamespace,
+			Environment: envMap,
+		}
+
+		envMap["RAW_BODY"] = "true"
+
+		stackYaml := &stack.Services{
+			Version: "1.0",
+			Provider: stack.Provider{
+				Name: "openfaas",
+			},
+			Functions: function,
+		}
+
+		// Serialize the stackYaml struct to yamlData that can be written into a YAML file
+		yamlData, err := yaml.Marshal(stackYaml)
+		if err != nil {
+			t.Fatalf("Cannot serialize struct into yaml stack:\n%v", err)
+		}
+
+		// Create a temp stack YAML file in the current dir
+		tempFile, err := ioutil.TempFile("./", "test-stack.*.yaml")
+		if err != nil {
+			t.Fatalf("Cannot create a tmp stack yaml file:\n%v", err)
+		}
+		defer os.Remove(tempFile.Name())
+
+		// Write the serialized yamlData into the temp file
+		err = ioutil.WriteFile(tempFile.Name(), yamlData, 655)
+		if err != nil {
+			t.Fatalf("Cannot write to temp file at: %s\nError:\n%v", tempFile.Name(), err)
+		}
+
+		// A Mock HTTP server that
+		srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if r.Method == http.MethodPut {
+				err := json.NewDecoder(r.Body).Decode(&request)
+				if err != nil {
+					t.Fatalf("Error decoding request:\n%v", err)
+				}
+				if request.Namespace != testCase.expectedNamespace {
+					t.Fatalf("Requested Namespace from faas-cli's request is: %s\nExpected namespace: %s\n",
+						request.Namespace, testCase.expectedNamespace)
+				}
+				w.WriteHeader(http.StatusAccepted)
+			}
+		}))
+
+		defer srv.Close()
+
+		test.CaptureStdout(func() {
+			faasCmd.SetArgs([]string{
+				"deploy",
```

Comment:
I am surprised this works, is the test server on localhost:8080 by default? it seems like you need to pass a gateway value here

Reply:
@LucasRoesler Yes you're right! I should pass in a gateway flag but as I pasted the output above, it seems to work fine, I'll update the faas-cli command to add the gateway flag and see if this fixes the error in the CI

Reply:
`srv` should have a URL you can use.

Reply:
This example may help https://github.com/alexellis/derek/blob/03cfe0757d50cc38c92f3823f4a53b5adff26581/auth/isCustomer_test.go#L41

Reply:
that url was already added 
![image](https://user-images.githubusercontent.com/891889/81596130-d3d3cb80-93c3-11ea-9ff9-60ecd46c415c.png)